internal/cortex/frontend/transport/handler.go

```diff
@@ -108,10 +108,10 @@ func NewHandler(cfg HandlerConfig, roundTripper http.RoundTripper, log log.Logge
 		_ = h.activeUsers.StartAsync(context.Background())
 	}
 
-	h.cachedHits = promauto.With(reg).NewCounter(prometheus.CounterOpts{
+	h.cachedHits = promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
 		Name: "cached_failed_queries_count",
 		Help: "Total number of queries that hit the failed query cache.",
-	})
+	}, []string{"total"})
```

Comment:
We don't need this label which doesn't provide any value. Can you try 
```
}, []string{})
```
and 
```
f.cachedHits.WithLabelValues().Inc()
```

Reply:
Can you move this counter into `utils/utils.go` and bump it there so that you can unit test it?

Reply:
addressed